hardhat/README.md

```diff
@@ -0,0 +1,236 @@
+# Using Optimistic Ethereum with the Hardhat Development Environment
+
+[![Discord](https://img.shields.io/discord/667044843901681675.svg?color=768AD4&label=discord&logo=https%3A%2F%2Fdiscordapp.com%2Fassets%2F8c9701b98ad4372b58f13fd9f65f966e.svg)](https://discord.com/channels/667044843901681675)
+[![Twitter Follow](https://img.shields.io/twitter/follow/optimismPBC.svg?label=optimismPBC&style=social)](https://twitter.com/optimismPBC)
+
+This tutorial aims to help you get started with developing decentralized applications on [Optimistic Ethereum](https://optimism.io/) using [Hardhat](https://hardhat.org/). Applications 
+running on top of Optimistic Ethereum are about as secure as those running on the underlying Ethereum mainnet itself, but are
+[significantly cheaper](https://optimism.io/gas-comparison).
+
+
+## Build an Optimistic Ethereum Server
+
+To test and debug on Optimistic Ethereum you need to have a running Optimistic Ethereum server, so the first step is to build one. The directions in this section are 
+for an Ubuntu 20.04 VM running on GCP with a 20 GB disk (the default, 10 GB, is not enough), but they should be similar for other Linux 
+versions and other platforms
+
+### Install Prerequisite Software
+
+1. Install [Docker](https://www.docker.com/). If you prefer not to use the convenience script, 
+   [there are other installation methods](https://docs.docker.com/engine/install/ubuntu).
+
+   ```sh
+   curl -fsSL https://get.docker.com -o get-docker.sh
+   sudo sh get-docker.sh
+   ```
+
+2. Configure Docker settings:
+
+   ```sh
+   sudo usermod -a -G docker `whoami`
+   sudo apt install -y docker-compose
+    ```
+
+3. Install [Node.js](https://nodejs.org/en/) and a number of npm packages. The version in the OS repository is 
+  out of date, so we'll get the package from a different source.
+
+   ```sh
+   curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
+   sudo bash nodesource_setup.sh
+   sudo apt install -y nodejs
+   ```
+   
+4. Install the Node.js packages we need.
+   ```sh   
+   sudo npm install -g yarn hardhat
+   ```
+   
+5. Log out and log back in to refresh the group information.
+
+
+### Start an Optimistic Ethereum Server
+
+This process downloads, compiles, and builds an Optimistic Ethereum network. Note that it takes a long time.
```

Comment:
```suggestion
This process downloads, compiles, and builds an Optimistic Ethereum network. Note that this may take a long time (up to 10 minutes).
```

Reply:
It took me more than that, so I'm reluctant to specify an amount of time.